processor/spanmetricsprocessor/processor.go

```diff
@@ -59,10 +62,18 @@ type exemplarData struct {
 	value   float64
 }
 
+// metricKey is used to carry the stringified metric attributes
 type metricKey string
 
+// resourceKey is used to carry the stringified resource attributes
+type resourceKey string
```

Comment:
They're not superfluous, they help to make sense of the structures that use them to construct multi-dimensional maps.  Rather than `map[string]map[string]int`, there is `map[resourceKey]map[metricKey]int`.

Reply:
@MovieStoreGuy https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7075#discussion_r781674403